sdk/src/azure/core/az_precondition.c

```diff
@@ -1,19 +1,18 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-#include <azure/core/az_platform.h>
 #include <azure/core/internal/az_precondition_internal.h>
-
 #include <stdint.h>
 
 #include <azure/core/_az_cfg.h>
 
 static void az_precondition_failed_default()
 {
-  /* By default, when a precondition fails the calling thread is suspended forever */
-  while (true)
+  /* By default, when a precondition fails the calling thread spins forever */
+  static uint16_t value = 0;
```

Comment:
Technical reason for not just a `while(1)`?

Reply:
clang -o2 results in the loop optimized away completely.